libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -258,6 +260,8 @@ class MapboxNavigation(
      */
     val historyRecorder = MapboxHistoryRecorder(navigationOptions, logger)
 
+    private var reachabilityObserverId: Long? = null
```

Comment:
```suggestion
    private val reachabilityObserverId: Long
```

The observer is added in the `init` block so we should be able to make it a value.

Reply:
We're checking for `null` 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/ca0f9c4ab5729403ac09af42be30b60c49efdf3b/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L283

https://github.com/mapbox/mapbox-navigation-android/blob/ca0f9c4ab5729403ac09af42be30b60c49efdf3b/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L518

I think that's not an option.

Reply:
@Guardiola31337 this does the trick. Just a nit, not a blocker.
```diff
diff --git a/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt b/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt
index 645700d7f..49307b137 100644
--- a/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt
+++ b/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt
@@ -260,7 +260,7 @@ class MapboxNavigation(
      */
     val historyRecorder = MapboxHistoryRecorder(navigationOptions, logger)
 
-    private var reachabilityObserverId: Long? = null
+    private val reachabilityObserverId: Long
 
     init {
         ThreadController.init()
@@ -280,11 +280,9 @@ class MapboxNavigation(
             MapboxModuleProvider.createModule(MapboxModuleType.NavigationRouter, ::paramsProvider),
             logger
         )
-        if (reachabilityObserverId == null) {
-            reachabilityObserverId = ReachabilityService.addReachabilityObserver(
-                connectivityHandler
-            )
-        }
+        reachabilityObserverId = ReachabilityService.addReachabilityObserver(
+            connectivityHandler
+        )
         directionsSession.registerRoutesObserver(navigationSession)
         val notification: TripNotification = MapboxModuleProvider
             .createModule(MapboxModuleType.NavigationTripNotification, ::paramsProvider)
@@ -515,10 +513,7 @@ class MapboxNavigation(
         MapboxNavigationTelemetry.unregisterListeners(this@MapboxNavigation)
         ThreadController.cancelAllNonUICoroutines()
         ThreadController.cancelAllUICoroutines()
-        ifNonNull(reachabilityObserverId) {
-            ReachabilityService.removeReachabilityObserver(it)
-            reachabilityObserverId = null
-        }
+        ReachabilityService.removeReachabilityObserver(reachabilityObserverId)
     }
 
     /**
```

Reply:
Ohh yeah, I wanted to prevent any bad usage when creating / destroying `MapboxNavigation`. Making sure there's always only 1 active `ReachabilityObserver`.